d20/tests/test_Registration.py

```diff
@@ -177,3 +181,433 @@ def test2registerverification(self):
         self.assertIn(
             "RegTestBackStory1",
             [backstory.registration.name for backstory in backstories],)
+
+
+def testRegistrationFormWrongOption():
+    with pytest.raises(TypeError) as excinfo:
+        @registerNPC(
+            name="Tester1",
+            description="Test NPC",
+            creator="",
+            version="0.1",
+            engine_version="0.1",
+            options="abc",
+        )
+        class RegTestNPC(NPCTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def handleData(self, **kwargs):
+                pass
+
+    assert str(excinfo.value) == "'options must be of type 'Arguments'"
+
+
+def testRegistrationFormWrongInterest():
+    with pytest.raises(ValueError) as excinfo:
+        @registerNPC(
+            name="Tester2",
+            description="Test NPC",
+            creator="",
+            version="0.1",
+            engine_version="0.1",
+            interests={'test': 'test'}
+        )
+        class RegTestNPC(NPCTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def handleData(self, **kwargs):
+                pass
+
+    assert str(excinfo.value) == "Unexpected keys in interests dict"
+
+    with pytest.raises(TypeError) as excinfo:
+        @registerNPC(
+            name="Tester2b",
+            description="Test NPC",
+            creator="",
+            version="0.1",
+            engine_version="0.1",
+            interests=1
+        )
+        class RegTestNPC2(NPCTemplate):
+            def __init__(self, **kwargs):
+                super().__init__(**kwargs)
+
+            def handleData(self, **kwargs):
+                pass
+
+    assert str(excinfo.value) == "Expected an interable type"
```

Comment:
we may want to change this to `iterable` and include a fix for the spelling in the code with the PR (bug in code)